aptos-move/move-examples/account_recovery/sources/account_recovery.move

```diff
@@ -0,0 +1,263 @@
+
+    const ERECOVERY_NOT_SET: u64 = 1;
+    const ERECOVERY_ALREADY_SET: u64 = 2;
+    const EUNAUTHORIZED: u64 = 3;
+    const ERECOVERY_NOT_IN_PROGRESS: u64 = 4;
+    const ERECOVERY_ALREADY_IN_PROGRESS: u64 = 5;
+    const EREVERSE_LOOKUP_NOT_INITIALIZED: u64 = 6;
+    const EINITIATOR_MISMATCH: u64 = 7;
+    const ERECOVERY_DELAY_NOT_MET: u64 = 8;
+
+    use std::signer;
+    use aptos_std::table::{Table, Self};
+    use aptos_framework::account;
+    use aptos_framework::timestamp;
+    use std::vector;
+    use std::option::Option;
+    use std::option;
+    use aptos_framework::resource_account;
+    use aptos_framework::account::SignerCapability;
+
+    /// Account recovery struct
+    struct AccountRecovery has key, store, drop {
+        authorized_addresses: vector<address>,
+        required_num_for_recovery: u64,
+        required_delay_seconds: u64,
+        rotate_valid_window_seconds: u64,
+        allow_unauthorized_initiation: bool,
+        account_recovery_init: Option<AccountRecoveryInitData>,
+    }
+
+    // Reverse lookup map
+    struct AccountRecoveryReverseLookup has key, store {
+        authorized_to_recovery: Table<address, vector<address>>,
+    }
+
+    struct AccountRecoveryInitData has key, store, drop {
+        initiator: address,
+        authorized: vector<address>,
+        recovery_seq_number: u64,
+        recovery_initiation_ts: u64,
+    }
+
+    struct ModuleData has key {
+        // Storing the signer capability here, so the module can programmatically sign for transactions
+        signer_cap: SignerCapability,
+    }
+
+    fun init_module(signer: &signer) {
+        move_to(signer, AccountRecoveryReverseLookup {
+            authorized_to_recovery: table::new(),
+        });
+
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(signer, @source_addr);
+
+        move_to(signer, ModuleData {
+            signer_cap: resource_signer_cap,
+        });
+    }
+
+    fun recovery_exists(addr: address): bool {
+        exists<AccountRecovery>(addr)
+    }
+
+    public entry fun register_without_authorization(
+        account: &signer,
+        required_delay_seconds: u64,
+        rotation_capability_sig_bytes: vector<u8>,
+        account_public_key_bytes: vector<u8>,
+    ) acquires AccountRecoveryReverseLookup {
+        register(account, vector::empty(), 0, required_delay_seconds, required_delay_seconds, true, rotation_capability_sig_bytes, account_public_key_bytes);
+    }
+
+    public entry fun register_authorize_one(
+        account: &signer,
+        authorized_address: address,
+        required_delay_seconds: u64,
+        rotation_capability_sig_bytes: vector<u8>,
+        account_public_key_bytes: vector<u8>,
+    ) acquires AccountRecoveryReverseLookup {
+        register(account, vector::singleton(authorized_address), 1, required_delay_seconds, required_delay_seconds, false, rotation_capability_sig_bytes, account_public_key_bytes);
+    }
```

Comment:
@sitalkedia if you have some time can you make this function either take in `rotate_valid_window_seconds` or make that really long?